src/rpc/net.cpp

```diff
@@ -1014,52 +1015,107 @@ static RPCHelpMan sendmsgtopeer()
-                              obj.pushKV("total", node.addrman->Size(network));
-                              ret.pushKV(GetNetworkName(network), obj);
-                          }
-                          UniValue obj(UniValue::VOBJ);
-                          obj.pushKV("new", node.addrman->Size(std::nullopt, true));
-                          obj.pushKV("tried", node.addrman->Size(std::nullopt, false));
-                          obj.pushKV("total", node.addrman->Size());
-                          ret.pushKV("all_networks", obj);
-                          return ret;
-                      },
+        "\nProvides information about the node's address manager by returning the number of "
+        "addresses in the `new` and `tried` tables and their sum for all networks.\n"
+        "When setting the verbose flag, the address manager entries for the new "
+        "and tried table are also returned.\n"
+        "This RPC is for testing only.\n",
+        {
+            {"verbose", RPCArg::Type::BOOL, RPCArg::Default{false}, "True for a list of address manager entries by table, false for address counts only"},
+        },
+        {
+            RPCResult{"for verbose = false", RPCResult::Type::OBJ_DYN, "", "json object with network type as keys",
+                {
+                    {RPCResult::Type::OBJ, "network", "the network (" + Join(GetNetworkNames(), ", ") + ", all_networks )", {
+                        {RPCResult::Type::NUM, "new", "number of addresses in the new table, which represent potential peers the node has discovered but hasn't yet successfully connected to."},
+                        {RPCResult::Type::NUM, "tried", "number of addresses in the tried table, which represent peers the node has successfully connected to in the past."},
+                        {RPCResult::Type::NUM, "total", "total number of addresses in both new/tried tables"},
+                    }},
+                },
+            },
+            RPCResult{"for verbose = true", RPCResult::Type::OBJ_DYN, "", "json object with network type, new_table and tried_table as keys",
+                {
+                    {RPCResult::Type::ELISION, "", "The same output as verbose = false"},
+                    {
+                        RPCResult::Type::ARR, "table", "list of addresses in the address manager table ( new_table, tried_table )",
+                        {
+                            {RPCResult::Type::OBJ, "", "an address manager table entry", {
+                                {RPCResult::Type::STR, "address", "the address"},
+                                {RPCResult::Type::NUM, "services", "the services the node might support"},
+                                {RPCResult::Type::NUM, "bucket", "the address manager bucket the address is placed in"},
+                                {RPCResult::Type::NUM, "position", "the bucket position the address is placed in"},
```

Comment:
Could substantially reduce the size of the json output (by about 20%) by changing this to:

```json
{
  "new_table": {
    "0/2": {
      "address": "89.78.111.197:8333",
      "services": 1037,
      "source": "34.126.67.135"
    },
    "0/5": {
      "address": "13.228.211.83:8333",
      "services": 1033,
      "source": "34.126.67.135"
    }
}
```

Reply:
TIL UniValue doesn't support integer keys - only string keys. Still going with this as it's about a 15% size reduction with somewhat empty addrman tables. I assume even more space savings on a full addrman tables, as the output would contain more `bucket` keys.

```
$ ll getrawaddrman.json getaddrmaninfo_verbose.json -h
-rw-r--r-- 1 user group 761K Sep 28 13:37 getrawaddrman.json
-rw-r--r-- 1 user group 881K Sep 28 13:37 getaddrmaninfo_verbose.json
```

Reply:
JSON in general  doesn't support integer keys.

Not a strong opinion, but there's two advantages of `"0/5":` vs `"0": {"5":` -- one is that if you don't care about the bucket/position, you can just skip through the values directly (`x.values()` vs `sum(x2.values() for x2 in x.values())`), the other is that bucket+position are pretty internal arrangements; if we end up with a different layout in the future, we can just map that to a different string template (eg "0/5/2") and only the people who care about the internals have to even notice the change.